### ISSUE TITLE
[BUG] Cannot pack with icon option

### ISSUE THREAD

**Issue by ner00** [dashingsoft/pyarmor#1749]
I've read the documentation here: https://pyarmor.readthedocs.io/en/latest/topic/repack.html#using-more-pyinstaller-options

Unless I failed to comprehend, I tried to specify an icon using the following command: 

**Setting icon argument**
```
C:\Python39>pyarmor cfg pack:pyi_options = "-i myicon.ico"
INFO     Python 3.9.4
INFO     Pyarmor 8.5.7 (trial), 000000, non-profits
INFO     Platform windows.x86_64
INFO     change option "pyi_options" to new value "-i myicon.ico"

------------------------------------------------------------
Section: pack

Current settings
  pyi_options = -i myicon.ico

Global settings

Local settings
  pyi_options = -i myicon.ico
```

**pyarmor.error.log**
```
11224 MainProcess 2024-04-23 18:45:56,569
Traceback (most recent call last):
  File "C:\Users\ner0\AppData\Roaming\Python\Python39\site-packages\pyarmor\cli\__main__.py", line 769, in main
    main_entry(sys.argv[1:])
  File "C:\Users\ner0\AppData\Roaming\Python\Python39\site-packages\pyarmor\cli\__main__.py", line 757, in main_entry
    return args.func(ctx, args)
  File "C:\Users\ner0\AppData\Roaming\Python\Python39\site-packages\pyarmor\cli\__main__.py", line 239, in cmd_gen
    builder.process(options, packer)
  File "C:\Users\ner0\AppData\Roaming\Python\Python39\site-packages\pyarmor\cli\generate.py", line 163, in process
    auto_resources = packer.analysis()
  File "C:\Users\ner0\AppData\Roaming\Python\Python39\site-packages\pyarmor\cli\repack.py", line 562, in analysis
    check_call(cmdlist, stdout=DEVNULL, stderr=DEVNULL)
  File "C:\Python39\lib\subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['C:\\Python39\\python.exe', '-m', 'PyInstaller', '--clean', '--workpath', '.pyarmor\\pack\\build', '-i', 'myicon.ico', '.pyarmor\\pack\\myscript.spec']' returned non-zero exit status 1.
```

**console output**
```
C:\Python39>pyarmor -d gen --pack onefile myscript.py
INFO     Python 3.9.4
INFO     Pyarmor 8.5.7 (trial), 000000, non-profits
INFO     Platform windows.x86_64
DEBUG    native platform windows.amd64
DEBUG    home path: C:\Users\ner0\.pyarmor
INFO     implicitly save obfuscated scripts to ".pyarmor\pack\dist"
DEBUG    command options: {'no_runtime': False, 'inputs': ['myscript.py'], 'output': '.pyarmor\\pack\\dist'}
DEBUG    install plugin: CodesignPlugin
DEBUG    install plugin: DarwinUniversalPlugin
DEBUG    Loaded ctypes backend
INFO     search inputs ...
INFO     find script myscript.py
INFO     find 1 top resources
DEBUG    C:\Python39\python.exe -m PyInstaller.utils.cliutils.makespec --specpath .pyarmor\pack -i myicon.ico myscript.py
INFO     call PyInstaller to generate specfile
DEBUG    C:\Python39\python.exe -m PyInstaller --clean --workpath .pyarmor\pack\build -i myicon.ico .pyarmor\pack\myscript.spec
INFO     call PyInstaller to analysis, it may take several minutes ...
DEBUG    unknown error, please check pyarmor.error.log
ERROR    Command '['C:\\Python39\\python.exe', '-m', 'PyInstaller', '--clean', '--workpath', '.pyarmor\\pack\\build', '-i', 'myicon.ico', '.pyarmor\\pack\\myscript.spec']' returned non-zero exit status 1.
ERROR    somthing is wrong
```

**Running manually**
```
C:\Python39>C:\Python39\python.exe -m PyInstaller --clean --workpath .pyarmor\pack\build -i myicon.ico .pyarmor\pack\myscript.spec
499 INFO: PyInstaller: 6.6.0, contrib hooks: 2024.4
499 INFO: Python: 3.9.4
500 INFO: Platform: Windows-10-10.0.19041-SP0
option(s) not allowed:
  --icon
makespec options not valid when a .spec file is given
```

**Note:** Running the same command without pyarmor packs fine with the intended icon `C:\Python39\python.exe -m PyInstaller --clean --workpath .pyarmor\pack\build -i myicon.ico myscript.py`

**Comment by ner00** [comment]
May be relevant: https://github.com/pyinstaller/pyinstaller/issues/7203#issuecomment-1295787184

**Comment by jondy** [comment]
No, it's pyarmor pack bug. I have fixed it, here it's pre-release version

https://pyarmor.dashingsoft.com/downloads/temp/pyarmor-8.5.7.zip

**Comment by jondy** [comment]
Fixed in 8.5.7
